coldfront/core/project/tests/test_commands/test_add_service_units_to_project.py

```diff
@@ -1,129 +1,63 @@
-            name="Service Units")
-        for project_user in project.projectuser_set.all():
-            if project_user.role.name != 'User':
-                continue
-
-            allocation_user = \
-                allocation_objects.allocation.allocationuser_set.get(
-                    user=project_user.user)
-            allocation_user_attribute = \
-                allocation_user.allocationuserattribute_set.get(
-                    allocation_attribute_type=allocation_attribute_type,
-                    allocation=allocation_objects.allocation)
-            historical_allocation_user_attribute = \
-                allocation_user_attribute.history.latest("id")
-            historical_reason = \
-                historical_allocation_user_attribute.history_change_reason
-            self.assertEqual(historical_reason,
-                             'This is a test for add_service_units command')
+        self.reason = 'This is a test for add_service_units command'
+
+        # Create Projects and associate Users with them.
+        project_status = ProjectStatusChoice.objects.get(name='Active')
+        project_user_status = ProjectUserStatusChoice.objects.get(
+            name='Active')
+        user_role = ProjectUserRoleChoice.objects.get(name='User')
+        manager_role = ProjectUserRoleChoice.objects.get(name='Manager')
+
+        for i in range(2):
+            # Create an ICA Project and ProjectUsers.
+            project = Project.objects.create(
+                name=f'project{i}', status=project_status)
+            setattr(self, f'project{i}', project)
+            for j in range(2):
+                ProjectUser.objects.create(
+                    user=getattr(self, f'user{j}'), project=project,
+                    role=user_role, status=project_user_status)
+            ProjectUser.objects.create(
+                user=self.pi, project=project, role=manager_role,
+                status=project_user_status)
+
+            # Create a compute allocation for the Project.
+            allocation = Decimal(f'{i + 1}000.00')
+            create_project_allocation(project, allocation)
+
+            # Create a compute allocation for each User on the Project.
+            for j in range(2):
+                create_user_project_allocation(
+                    getattr(self, f'user{j}'), project, allocation / 2)
 
     def test_dry_run(self):
         """Testing add_service_units_to_project dry run"""
-        out, err = StringIO(''), StringIO('')
-        call_command('add_service_units_to_project',
-                     '--project_name=project0',
-                     '--amount=1000',
-                     '--reason=This is a test for add_service_units command',
-                     '--dry_run',
-                     stdout=out,
-                     stderr=err)
-        sys.stdout = sys.__stdout__
-        out.seek(0)
+        output, error = self.call_deactivate_command('add_service_units_to_project',
```

Comment:
You might just put `call_command` in the test class, as I mentioned in #358. (This is mainly confusing because of the method name.)

Reply:
Yup. overlooked that